### ISSUE TITLE
Pressure level weights computation issue?

### ISSUE THREAD

**Issue by danielabdi-noaa** [google-deepmind/graphcast#156]
In the GraphCast code that computes weights for each pressure level in `losses.py/normalized_level_weights` function

https://github.com/google-deepmind/graphcast/blob/7077d40a36db6541e3ed72ccaed1c0d202fa6014/graphcast/losses.py#L97-L100

the `mean() ` is used to normalize instead of `sum()` , which I thought was the correct way according to Figure 6 in the GraphCast paper and the following statement

> The loss weights across all variables sum to 7.4, i.e., (6 × 1.0 for the atmospheric variables, plus (1.0 + 0.1 + 0.1 + 0.1 + 0.1) for the surface variables listed above, respectively).

However, when using `mean()` to normalize the weights, it sums upto the number of pressure levels (37?) instead of 1. This would have the undesired effect of reducing the weights for the four surface variables by a large margin!

I am sure that I am missing something -- maybe this is corrected in the code somewhere else?.

**Comment by alvarosg** [comment]
Thanks for your message.

Note the reasons the weights are normalized by the mean, is because then, when actually reducing the "level" dimension (which gets reduced [here](https://source.corp.google.com/piper///depot/google3/third_party/py/graphcast/losses.py;rcl=574127233;l=73)), we also use a "mean", instead of a "sum".

E.g. we do:

```
level_weights / level_weights.mean(skipna=False)
loss_with_level = loss_with_level * level_weights
loss_with_level_reduced = loss_with_level.mean("level")
```

For a variable without level none of this applies, and you would simply get:
```
loss_surface_var
```

When you add those two together:
you get:

```
total_loss = loss_with_level_reduced + surface_var_weight * loss_surface_var
```
where:

`surface_var_weight` takes a value of `1.` for `2t` or `0.1` for the others.

This means that `2t`, is weights as much as *all of the levels of a pressure level variable combined*, while the others surface vars get 1/10. of that.


We could have chosen instead do something like:

```
level_weights / level_weights.sum(skipna=False)
loss_with_level = loss_with_level * level_weights
loss_with_level_reduced = loss_with_level.sum("level")
```

and believe the result would have been the same, and be a bit more intuitive. However working with makes the scales of the intermediate arrays a bit easier to interpret (E.g. if we log loss per level).

Hope this helps!

**Comment by danielabdi-noaa** [comment]
Thank you @alvarosg ! 
Yes, that makes total sense now. Sorry I thought that `_mean_preserving_batch` computes average over 
the batch when in fact it averages over all dimensions  except the "batch" dimension.
